### PR TITLE
Unboxed Bits8/Bits16 using Int as the underlying representation

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

**Comment by melted** [comment]
On first glance it looks good. I'll do a proper review soon.

File: rts/idris_gc.c [hunk 1]
```diff
@@ -4,17 +4,27 @@
 #include "idris_bitstring.h"
 #include <assert.h>
 
+static inline VAL plain(VM* vm, VAL x, size_t sz) {
```
Comment (melted) [inline rts/idris_gc.c]: call this function `copy_plain`

File: rts/idris_rts.c [hunk 2]
```diff
@@ -978,7 +965,7 @@ Msg* idris_checkInitMessages(VM* vm) {
     Msg* msg;
 
     for (msg = vm->inbox; msg < vm->inbox_end && msg->msg != NULL; ++msg) {
-	if ((msg->channel_id & 1) == 1) { // init bit set
+        if (msg->channel_id && 1 == 1) { // init bit set
```
Comment (melted) [inline rts/idris_rts.c]: this change is wrong, it should be bitwise and.

File: rts/idris_rts.c [hunk 3]
```diff
@@ -417,11 +409,10 @@ void dumpStack(VM* vm) {
 void dumpVal(VAL v) {
     if (v==NULL) return;
     int i;
-    if (ISINT(v)) {
-        printf("%d ", (int)(GETINT(v)));
-        return;
-    }
     switch(GETTY(v)) {
+    case CT_INT:
+        printf("%d ", (int)(GETINT(v)));
```
Comment (melted) [inline rts/idris_rts.c]: This was wrong before, but this always prints an int as a 32-bit number, might as well fix that.

File: rts/idris_rts.c (hunk 2 above)
Comment (jacereda) [reply rts/idris_rts.c]: Oops, this PR is based on another branch of mine that didn't contain some of my other fixes, sorry, let me fix the mess.

**Comment by melted** [comment]
Looks good! Thanks!

**Pull request merged**